### DISCUSSION TITLE
Logs are not shown in structured logs

### DISCUSSION THREAD

**Discussion by halilkocaoz** [microsoft/aspire#1908] [Q&A]
I use the [LoggerFactory](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerfactory?view=dotnet-plat-ext-8.0) class to create logger.
Like this:

```
public class ClassName<TValue> : Interface<TValue>
{
    private readonly ILogger<ClassName<TValue>> _logger;
   
    public ClassName()
    {
        _logger = LoggerFactory
            .Create(builder => builder.AddConsole())
            .CreateLogger<ClassName<TValue>>();
    }
}
```

In the Aspire dashboard I only see the logs from the console and I have checked the builder but I can't find anything to help. Do you have any idea why the logs are not showing up in the structured logs?

**Comment by davidfowl** [comment]
You create a new logger factory with no open telemetry configured. Take a look at the ServiceDefaults. That's how logging (metrics and tracing) is wired up to the dashboard.

Why aren't you just injecting a logger into your class?

**Reply by halilkocaoz** [reply]
I thought that's open telemetry, but I couldn't find a solution configure open telemetry with LoggerFactory.

There are types in the ClassName constructor that cannot be resolved by the DI container like the primitive types, probably i will pass the ILogger with the constructor.
